tests/suites/hup/tests/rollback-altboot.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 balena
+ *
+ * @license Apache-2.0
+ */
+
+'use strict';
+
+module.exports = {
+  title: 'Rollback altboot (broken init) test',
+  run: async function(test) {
+    await this.context.get().hup.initDUT(
+      this, test, this.context.get().link
+    );
+
+    await this.context.get().hup.doHUP(this, test, 'image', this.context.get().hup.payload, this.context.get().link);
+
+    await this.context.get().worker.executeCommandInHostOS(
+      `sed -i -e "s/COUNT=.*/COUNT=2/g" -e "s/TIMEOUT=.*/TIMEOUT=60/g" $(find /mnt/sysroot/inactive/ | grep "bin/rollback-altboot")`,
+      this.context.get().link,
+    );
+
+    // break init
+    await this.context.get().worker.executeCommandInHostOS(
+      `rm /mnt/sysroot/inactive/current/boot/init`,
+      this.context.get().link,
+    );
+
+    await this.context.get().worker.rebootDut(this.context.get().link);
```

Comment:
If altboot is triggered, the DUT will actually reboot again after reverting the boot files. So we actually need to expect another reboot after this one.

We could try and connect between the reboots and see if the rollback services are running, but if we miss the window it may trigger a false-positive.